### DISCUSSION TITLE
Are there any projects that use React-RxJS?

### DISCUSSION THREAD

**Discussion by pastelmind** [re-rxjs/react-rxjs#288] [Q&A] [answered]
Hi! I am impressed by your library. I'm looking to use React and RxJS in my project, and I want to learn by examples. 

I checked the example repos under re-rxjs: [react-rxjs-basic-todos](https://github.com/re-rxjs/react-rxjs-basic-todos), [react-rxjs-github-issues-example](https://github.com/re-rxjs/react-rxjs-github-issues-example), and [eleccions-2021](https://github.com/re-rxjs/eleccions-2021). But these were fairly basic CRUD projects. I'm looking for apps that use non-trivial observable chains, and that handle errors in a sensible way.

Are there any (hopefully large and mature) projects that use React-RxJS?

Edit: I forgot there was a Discussions tab--please move this issue over there if need be.

**Comment by josepot** [comment, marked as answer]
Hey!

Sorry for the late answer.

AFAIK, there are in fact a few large and mature projects using React-RxJS:

- At [Adaptive financial consulting](https://weareadaptive.com/) they are using it in a few large projects for building complex trading UIs, and AFAIK they are quite happy with the library. Unfortunately, though, those projects are not open source.

- At [Contango](https://contango.exchange/) they are using it for building their dApp, which is also a trading UI. I'm pretty sure that their UI will be open sourced in the coming weeks.

- At [tripleplay](https://tripleplay.ai/) (formerly lobby) they also used it for building their complex video UI. Although, I'm not sure if they kept using the library after they got acquired, I will ask them.

- @voliva  and myself used it for creating a [quite successful PWA](https://github.com/re-rxjs/eleccions-2021) that was used for following the Catalan elections in real time. BTW, this is not a very complex project (as we hacked it in 10 days or so), but IMO it's definitely far from being a basic CRUD project. 

That being said, this library has its trade-offs: the main one being that its not suited for SSR. However, for creating complex SPAs that have to deal with real-time data, that require for the UI to derive a lot of state on the client. Then, as long as SSR is not a must, for those kinds of UIs is IMO much better than using other state-management libraries... But ofc I'm super biased :slightly_smiling_face: 

@voliva and I have been using it professionally for the last 2-3 years for building some pretty cool projects, and after these years we have reached the conclusion that perhaps another set of trade-offs would be more desirable. So, that we could enable for SSR and context state, without having to worry about unsubscriptions...  That's why we are now working on a new (and possibly better) RxJS-based state-management library.

However, we try to keep these developments as "low-key" as we can,  for a number of reasons... I mean, we build the libraries to solve problems that we face in a way to fits our mental-models, but we have not intention of becoming zealots/advocates/preachers of these mental-models... That being said, if you find these mental-models interesting, here we are to share and learn together, for sure!
